queenbee/recipe/recipe.py

```diff
@@ -682,17 +682,24 @@ def template_by_name(templates: List[Union[DAG, TemplateFunction]], name: str) -
         return res
 
 
-class RecipeInfo(BaseModel):
-    """Recipe information.
+class RecipeInterface(BaseModel):
+    """An interface object for creating a Recipe.
 
-    Recipe information only includes metadata, inputs and outputs of a Recipe.
+    Recipe information only includes metadata, source, inputs and outputs of a Recipe.
     This object is useful for creating user interface for Recipes.
     """
+    type: constr(regex='^RecipeInterface$') = 'RecipeInterface'
+
     metadata: MetaData = Field(
         ...,
         description='Recipe metadata information.'
     )
 
+    source: str = Field(
```

Comment:
Reckon we can use the [`AnyUrl`](https://pydantic-docs.helpmanual.io/usage/types/#urls) type from Pydantic here maybe?

Reply:
I like using a more rigid type but AnyUrl doesn't allow `file:///` format.